ee/server/service/service_teams.go

```diff
@@ -286,3 +287,65 @@ func (svc *Service) ModifyTeamEnrollSecrets(ctx context.Context, teamID uint, se
 
 	return newSecrets, nil
 }
+
+func (svc Service) ApplyTeamSpecs(ctx context.Context, specs []*fleet.TeamSpec) error {
+	if err := svc.authz.Authorize(ctx, &fleet.Team{}, fleet.ActionRead); err != nil {
+		return err
+	}
+
+	config, err := svc.AppConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, spec := range specs {
+		var secrets []*fleet.EnrollSecret
+		for _, secret := range spec.Secrets {
+			secrets = append(secrets, &fleet.EnrollSecret{
+				Secret: secret.Secret,
+			})
+		}
+
+		team, err := svc.ds.TeamByName(ctx, spec.Name)
+		if err != nil {
+			if err := ctxerr.Cause(err); err == sql.ErrNoRows {
+				agentOptions := spec.AgentOptions
+				if agentOptions == nil {
+					agentOptions = config.AgentOptions
+				}
+				_, err = svc.ds.NewTeam(ctx, &fleet.Team{
+					Name:         spec.Name,
+					AgentOptions: agentOptions,
+					Secrets:      secrets,
+				})
+				if err != nil {
+					return err
+				}
+				continue
+			}
+
+			return err
+		}
+
+		// See if the user can write to that team in particular
```

Comment:
Maybe for loop iterates and fails the whole request if one of the teams is not authorized.

Then a second for loop to `SaveTeam`+`ApplyEnrollSecrets`.